tutorial/chapter1.md

```diff
@@ -0,0 +1,111 @@
+
+int main(int argc, char* argv[]) {
+    std::string containerName = argc > 1 ? argv[1] : "sample1";
+    std::string folderName = argc > 4 ? argv[4] : TEST_DATA;
+    kpsr::mem::MemEnv environment;
+
+    kpsr::vision_ocv::high_performance::ImageMultiplexerVisionMiddlewareProvider<128> provider(nullptr, "image_processing_feed",
+                                                                                              1024, 640, 16, "body");
+
+    kpsr::EventEmitterMiddlewareProvider<kpsr_tutorial::ResultData> eventEmitter(nullptr,
+                                                                        "image_processing_results",
+                                                                        0, nullptr, nullptr);
+
+    SimpleWriteService simpleWriteService(&environment,
+                                          provider.underlying->getPublisher(),
+                                          folderName,
+                                          true);
+    kpsr_tutorial::ImageProcessingService imageProcessingService(&environment, provider.underlying->getSubscriber(), eventEmitter.getPublisher());
+
+    imageProcessingService.startup();
+    simpleWriteService.startup();
+
+    ServerUserInput serverUserInput;
+
+    std::thread imageProcessingServiceThread = std::thread([&]() {
+        while (true) {
+            simpleWriteService.runOnce();
+            imageProcessingService.runOnce();
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            if (!serverUserInput._isRunning) {
+                break;
+            }
+        }
+    });
+    serverUserInput.run();
+
+    imageProcessingServiceThread.join();
+
+    simpleWriteService.shutdown();
+    imageProcessingService.shutdown();
+}
+
+```
+In this example there are some features of Klepsydra that will be introduced and others that have been described in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md).
+
+* From the module kpsr-robotics/vision_ocv/modules/mem_mdlw, the ```kpsr::vision_ocv::ImageMultiplexerVisionMiddlewareProvider``` class is introduced. Note that a param of this class is ```kpsr::high_performance::DataMultiplexerMiddlewareProvider```, which is a single-producer, multiple-consumer introduced in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md).
+
+* From the module kpsr-robotics/vision_ocv/modules/vision_ocv, ```kpsr::vision_ocv::ImageDataFactory```class is introduced. It is used to handle the allocation of images, copy and resizing.
+
+* The ```kpsr::Service``` was introduced in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md) and it is used in this example (by means of inheritance) to create two custom services:
+
+   * ```SimpleWriteService```, which is a service that resizes and publishes the resized images
```

Comment:
simpleWriteService is to read images from a folder.

Reply:
In fact, ther FileImageStreamService from kpsr-robotics could be used here instead. Can you use that in the example instead?

Reply:
Changing it to spdlog::info there will print too many messages (you have to think about the case where it is being used in other projects). Better solution would be to set the log level to debug in the main.cpp of the tutorial (spdlog::set_log_level i think). That way all the debug messages are printed.

Reply:
Done